### PR TITLE
bugfix break infinite loop on aws billing

### PR DESCRIPTION
**Pull request by aspacca**
Bug
## What does this PR do?
Return from `billing. getAccountName` in case of error, breaking an infinite loop

## Why is it important?
The for loop will never return otherwise

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/metricbeat/module/aws/billing/billing.go [hunk 1]
```diff
@@ -444,8 +444,7 @@ func (m *MetricSet) getAccountName(svc *organizations.Client) map[string]string
 	for paginator.HasMorePages() {
 		page, err := paginator.NextPage(context.Background())
 		if err != nil {
-			//TODO continue or return with error? Probably is return but...
-			continue
+			return accounts
```
Comment (tommyers-elastic) [inline x-pack/metricbeat/module/aws/billing/billing.go:447]: do we not want to surface the error to the caller so they are aware the return value may not contain the full list of accounts?

Comment (aspacca) [reply x-pack/metricbeat/module/aws/billing/billing.go:447]: indeed we could add a log entry at warn level
when I tested the code the error I had was that I have no permission for the API call

"funny" enough that the paginator for an API call that I don't have permission to call tell us that "there are more pages"

**Comment by aspacca** [comment]
@tommyers-elastic I've addressed the comment

please let me know if it's good to merge

thanks

**Review by tommyers-elastic — APPROVED** [review]
thanks andrea

**Pull request merged**